src/transpiling/mod.rs

```diff
@@ -290,11 +292,15 @@ pub fn fold_program(
   let mut passes = chain!(
     Optional::new(transforms::StripExportsFolder, options.var_decl_imports),
     resolver(unresolved_mark, top_level_mark, true),
-    proposal::decorators::decorators(proposal::decorators::Config {
-      legacy: true,
-      emit_metadata: options.emit_metadata,
-      use_define_for_class_fields: true,
-    }),
+    Optional::new(
+      proposal::decorators::decorators(proposal::decorators::Config {
+        legacy: true,
+        emit_metadata: options.emit_metadata,
+  
+        use_define_for_class_fields: true,
+      }),
+      !options.use_decorator_proposal,
+    ),
```

Comment:
Backwards compat - the TS experiemental decorators transform is only applied if `use_decorator_propsal` hasn't been configured